packages/deployments/contracts/contracts/lib/Connext/ConnextLogic.sol

```diff
@@ -864,6 +878,177 @@ library ConnextLogic {
+   * @return The amount left after the repayment
+   * @return The address of the adopted asset
+   */
+  function _reconcileProcessPortal(
+    ReconcileArgs memory _args,
+    uint256 _amount,
+    address _token,
+    bytes32 _transferId,
+    mapping(bytes32 => uint256) storage _aavePortalsTransfers,
+    mapping(bytes32 => IStableSwap) storage _adoptedToLocalPools,
+    mapping(bytes32 => address) storage _canonicalToAdopted
+  ) private returns (uint256, address) {
+    ReconcilePortalVars memory vars;
+
+    vars.portalTransferAmount = _aavePortalsTransfers[_transferId];
+
+    if (vars.portalTransferAmount == 0) return (_amount, _token);
+
+    // Swap all the transfer amount from local asset to adopted if needed
+    (vars.adoptedAmountOut, vars.adopted) = AssetLogic.swapFromLocalAssetIfNeeded(
+      _canonicalToAdopted,
+      _adoptedToLocalPools,
+      _args.tokenRegistry,
+      _token,
+      _amount
+    );
+
+    // Calculates the amount to be repaid to the portal
+    (vars.totalRepayAmount, vars.backUnbackedAmount, vars.portalFee) = _calculatePortalRepayment(
+      vars.adoptedAmountOut,
+      vars.portalTransferAmount,
+      _args.portalFeeNumerator,
+      _args.portalFeeDenominator
+    );
+
+    // Edge case with some tokens: Example USDT in ETH Mainnet, after the backUnbacked call there could be a remaining allowance if not the whole amount is pulled by aave.
+    // Later, if we try to increase the allowance it will fail. USDT demands if allowance is not 0, it has to be set to 0 first.
+    // TODO: Should we call approve(0) and approve(totalRepayAmount) instead? or with a try catch to not affect gas on all cases?
+    // Example: https://github.com/aave/aave-v3-periphery/blob/ca184e5278bcbc10d28c3dbbc604041d7cfac50b/contracts/adapters/paraswap/ParaSwapRepayAdapter.sol#L138-L140
+    SafeERC20Upgradeable.safeIncreaseAllowance(IERC20Upgradeable(vars.adopted), _args.aavePool, vars.totalRepayAmount);
+
+    // TODO: What if backUnbacked call fails? For example if the asset is inactive, paused or frozen?
+    // What is the impact of a reconcile to be reverted? Can be re-executed later?
+    IAavePool(_args.aavePool).backUnbacked(vars.adopted, vars.backUnbackedAmount, vars.portalFee);
+
+    // TODO should emit some event with the repayment info?
+
+    // TODO: Do we need to check balance before and after to get the exact amount paid and give the routers the rest?
+    // Aave accounts a global unbacked variable per asset for all, not by address/bridge.
+    // Someone can repay more than it should, so then a the moment of calling backUnbacked()
+    // aave can pull a smaller amount than backUnbackedAmount. So there will be an extra amount of assets that needs to be assigned
+    // See https://github.com/aave/aave-v3-core/blob/feb3f20885c73025f40cc272b59e7eacfaa02fe4/contracts/protocol/libraries/logic/BridgeLogic.sol#L121
```

Comment:
if we think this will be more than dust, we should probably assign it. however, its not clear to me where the excess would come from (i.e. who repaid more than they should). 

open to just distributing to all the routers though and leaving it as luck of the draw

Reply:
> its not clear to me where the excess would come from (i.e. who repaid more than they should).

It could be any of our routers for example due to a miscalculation in a manual repayment on the separate method or could be any other bridge solution that is also using Aave Portals

Reply:
it seems like we can discount this then, i dont see it being likely more than dust and im not sure its worth the added complexity